crates/bevy_reflect/src/type_registry.rs

```diff
@@ -350,11 +350,103 @@ impl<T: for<'a> Deserialize<'a> + Reflect> FromType<T> for ReflectDeserialize {
     }
 }
 
+#[derive(Clone)]
+pub struct ReflectFromPtr {
```

Comment:
As someone who does not really understand Rust's pointer mechanics (😅), it would be helpful if this struct (and its methods) included more documentation as to the purpose of this struct and how it's used.

Reply:
I added some docs: https://github.com/bevyengine/bevy/pull/4475/commits/46651954388dc59ad752340413b6351f144b6c2b#diff-1eebcee1cf56cedd9ca4b25ba011606b671b7206a1705fddc669461faf3164feR353-R381

Reply:
Reading over it again I think it should also why `*const T -> *dyn Reflect` isn't a no-op (because it changes the pointer to a "fat" pointer pointing to both the value and the vtable of the implementation of `T` for `Reflect`)

Reply:
Awesome! These docs are really good and definitely helped me understand what was going on. Thanks for this!